### PR TITLE
Execute a pipeline

### PR DESCRIPTION
**Pull request by AGrigis**
Deal with optional parameters in order to execute pipelines from the gui.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,7 +23,6 @@ install:
     - pip install nibabel --use-mirrors
     - pip install nose-cov --use-mirrors
     - pip install nipype --use-mirrors
-    - pip install xmltodict --use-mirrors
```
Comment (sapetnioc) [inline .travis.yml]: Is this change related to the subject of the pull request ? I have the feeling that it should be in another pull request.

File: capsul/pipeline/pipeline.py [hunk 2]
```diff
@@ -61,11 +59,7 @@ def pipeline_definition(self):
               self.add_link('proc2.out1->main_switch.in2_switch_out1')
               self.add_link('proc2.out1->main_switch.in2_switch_out2')
 
-    After execution of :py:meth:`pipeline_definition`, the inner nodes
-    parameters which are not connected will be automatically exported to the
-    parent pipeline, with names prefixed with their process name, unless they
-    are listed in a special "do_not_export" list (passed to
-    :py:meth:`add_process` or stored in the pipeline instance).
+    After execution of :py:meth:`pipeline_definition`, the inner nodes parameters which are not connected will be automatically exported to the parent pipeline, with names prefixed with their process name, unless they are listed in a special "do_not_export" list (passed to :py:meth:`add_process` or stored in the pipeline instance).
```
Comment (sapetnioc) [inline capsul/pipeline/pipeline.py]: I think I am missing something here. This change (and other similar changes) is the contrary of PEP 8, no ?

File: capsul/pipeline/pipeline.py [hunk 3]
```diff
@@ -938,8 +924,8 @@ def update_nodes_and_plugs_activation(self):
                 if test:
                     for plug_name, plug in test:
                         if debug:
-                            print >> debug, '%d-%s:%s' % (
-                                iteration, node.full_name, plug_name)
+                            print >> debug, '%d-%s:%s' % (iteration, node.full_name, plug_name)
+                        #print '!deactivations! iteration', iteration, '--- %s:%s' % (node.full_name,plug_name)
```
Comment (sapetnioc) [inline capsul/pipeline/pipeline.py]: Why this commented print is inserted here. It looks like the kind of debug lines that I use sometimes. Is it a line that have been removed in a previous changeset and that is reintroduced here ?

File: capsul/plugins/__init__.py [hunk 4]
```diff
@@ -11,5 +11,5 @@
     ("caps", "http://nsap.intra.cea.fr/caps-doc/"),
     ("qmri", "https://bioproj.extra.cea.fr/redmine/projects/qmri/publishing/"),
     ("funtk", "https://bioproj.extra.cea.fr/redmine/projects/funtk/publishing/"),
-    ("pclinfmri", "")
```
Comment (sapetnioc) [inline capsul/plugins/__init__.py]: There is no documentation here. I do not know (or remember) how this module is used.

File: capsul/process/controller.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
+#! /usr/bin/env python
```
Comment (sapetnioc) [inline capsul/process/controller.py]: Why are controllers inserted here since they are in soma-base ?

**Comment by sapetnioc** [comment]
I think this pull request should be divided in several ones for the sake of the reviewing process. There are modification on 30 files and not all seem to be related to the topic of the pull request. Some modifications are cosmetics, some are GUI, some are about plug-ins, some are about controllers, etc. To my opinion, it is a bit messy.

**Comment by denisri** [comment]
I am also under the impression that this PR brings back a bunch of older code. But maybe it is just because it was based on an older state of the master branch ?
Anyway some of the changes should probably not be included, in the first place the controller.py, for sure, which is in soma-base.

**Comment by AGrigis** [comment]
Yes. That's why we have to merge this PR manually with caution.0

**Comment by AGrigis** [comment]
It can be merged.
